sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CbdeHandlerRule.cs

```diff
@@ -68,11 +67,12 @@ internal static CbdeHandlerRule MakeUnitTestInstance(string testCbdeBinaryPath,
             return new CbdeHandlerRule(true, testCbdeBinaryPath, onCbdeExecution);
         }
 
-        protected sealed override void Initialize(SonarAnalysisContext context)
+        protected override void Initialize(SonarAnalysisContext context)
         {
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            // The available platform ids are documented here: https://docs.microsoft.com/en-us/dotnet/api/system.platformid?view=netframework-4.8
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT)
```

Comment:
In the notes for #3048, the CFamily team said they tried to use `Environment.OSVersion.Platform` but it "didn't work on .NET Core". Perhaps they tried a different configuration where it failed? Or were they just using a version of .NET Core older than v2.0?

Reply:
I'm not sure why it didn't work for them. I've asked @loic-joly-sonarsource for more details since this seems to be supported starting with .Net Core 2: https://docs.microsoft.com/en-us/dotnet/api/system.environment.osversion?view=netframework-4.8